mrp_production_line_group/views/mrp_production.xml

```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="mrp_production_form_view" model="ir.ui.view">
+        <field name="name">mrp.production.form</field>
+        <field name="model">mrp.production</field>
+        <field name="inherit_id" ref="mrp.mrp_production_form_view"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='is_locked']" position="after">
+                <field name="lines_grouped" invisible="1"/>
+            </xpath>
+            <xpath expr="//header" position="inside">
+                <button type="action"
+                    name="%(mrp_production_line_group.action_production_group_line)d"
+                    string="Group Lines"
+                    attrs="{'invisible': ['|', ('state', 'not in', ('confirmed','planned')), ('id', '=', False), '|', ('lines_grouped', '=', True)]}"/>
+            </xpath>
+            <xpath expr="//button[@name='%(mrp.action_change_production_qty)d']" position="attributes">
+                <attribute name="attrs">{'invisible': ['|', ('lines_grouped', '=', True)]}</attribute>
```

Comment:
OR operator is useless in this case

Reply:
I misunderstood concatenation, put entire domain.